SIT.Manager.Avalonia/ViewModels/Installation/ConfigureSitViewModel.cs

```diff
@@ -0,0 +1,195 @@
+
+    [ObservableProperty]
+    private bool _hasMirrorsAvailable = false;
+
+    public ObservableCollection<SitInstallVersion> AvailableVersions { get; } = [];
+    public ObservableCollection<KeyValuePair<string, string>> AvailableMirrors { get; } = [];
+
+    public IAsyncRelayCommand ChangeEftInstallLocationCommand { get; }
+
+    public ConfigureSitViewModel(IManagerConfigService configService, IInstallerService installerService, IPickerDialogService pickerDialogService) : base()
+    {
+        _configService = configService;
+        _installerService = installerService;
+        _pickerDialogService = pickerDialogService;
+
+        ChangeEftInstallLocationCommand = new AsyncRelayCommand(ChangeEftInstallLocation);
+    }
+
+    private async Task ChangeEftInstallLocation()
+    {
+        IStorageFolder? directorySelected = await _pickerDialogService.GetDirectoryFromPickerAsync();
+        if (directorySelected != null)
+        {
+            if (directorySelected.Path.LocalPath == CurrentInstallProcessState.BsgInstallPath)
+            {
+                // TODO show an error of some kind as we don't want the legit install to be the same as the SIT install.
+            }
+            else
+            {
+                CurrentInstallProcessState.EftInstallPath = directorySelected.Path.LocalPath;
+                ValidateConfiguration();
+            }
+        }
+    }
+
+    /// <summary>
+    /// Fetch the available versions and the download mirrors for these versions so we can populate the necesarry selections
+    /// and save loading time later
+    /// </summary>
+    /// <returns></returns>
+    private async Task FetchVersionAndMirrorMatrix()
+    {
+        IsLoading = true;
+
+        // Clear the collections
+        HasVersionsAvailable = false;
+        HasMirrorsAvailable = false;
+
+        AvailableVersions.Clear();
+        AvailableMirrors.Clear();
+
+        List<SitInstallVersion> availableVersions = await _installerService.GetAvailableSitReleases(CurrentInstallProcessState.EftVersion);
+        if (!string.IsNullOrEmpty(_configService.Config.SitVersion))
+        {
+            availableVersions = availableVersions.Where(x =>
+            {
+                bool parsedSitVersion = Version.TryParse(x.SitVersion.Replace("StayInTarkov.Client-", ""), out Version? sitVersion);
+                if (parsedSitVersion)
+                {
+                    Version installedSit = Version.Parse(_configService.Config.SitVersion);
+                    if (sitVersion >= installedSit)
+                    {
+                        return true;
+                    }
+                }
+                return false;
+            }).ToList();
+        }
+
+        // Make sure we only offer versions which are actually available to use to maximize the chances the install will work
+        AvailableVersions.AddRange(availableVersions.Where(x => x.IsAvailable));
+        if (AvailableVersions.Any())
```

Comment:
I don't blame you for this because I SWEAR microsoft used to tell you to use `Any()` but now they recommend using `.Count > 0`